express-notes-api/index.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const notesJsonData = require('./data.json');
+let notes = Object.values(notesJsonData.notes);
+let nextId = notesJsonData.nextId;
+
+const jsonMiddleware = express.json();
+app.use(jsonMiddleware);
+
+app.get('/api/notes', (req, res, next) => {
+  res.status(200).json(notes);
+});
+
+app.get('/api/notes/:id', (req, res, next) => {
+  const reqId = parseInt(req.params.id, 10);
+  const note = notes.filter(x => x.id === reqId);
+  note.length
+    ? res.status(200).json(note[0])
+    : Math.floor(reqId) !== reqId || typeof reqId !== 'number'
+      ? res.status(400).json({ error: 'id must be a positive integer' })
```

Comment:
It is important to check that `id` is a good value (positive integer) _before_ we try to find the matching note. That way we don't to extra computation to search for the note unnecessarily.